SourceX/dx.cpp

```diff
@@ -251,7 +251,8 @@ void LimitFrameRate()
 	uint32_t v = 0;
 	if (frameDeadline > tc) {
 		v = tc % refreshDelay;
-		SDL_Delay(v / 1000 + 1); // ceil
+		if (sgOptions.Graphics.bFPSLimit)
+			SDL_Delay(v / 1000 + 1); // ceil

```

Comment:
Do an early return in the function instead of additionally nested ifs. We don't need to do any of the other calculations anyway.

Reply:
Like this?

if (frameDeadline > tc && sgOptions.Graphics.bFPSLimit) {
		v = tc % refreshDelay;
		SDL_Delay(v / 1000 + 1); // cei`

Reply:
No that is again nesting conditions.

early return:
```
if (!sgOptions.Graphics.bFPSLimit)
  return;
```